consensus/safety-rules/src/safety_rules.rs

```diff
@@ -158,6 +158,8 @@ impl<T: Payload> TSafetyRules<T> for SafetyRules<T> {
         }
     }
 
+    // @REVIEW Which of these TODOs are necessary for safety? Are all of the checks in this method
+    // strictly necessary for safety?
```

Comment:
strictly for safety? We honestly could strip out most of safety rules and just have a series of counters and the private key.

Reply:
I think the risk here is that if safety rules becomes "safety rules. and some other checks" that either (1) people we duplicate all other checks outside of safety rules (2) in order to avoid duplication people will minimize the situation in (1) but this risks people misunderstanding what checks are in safety rules.

Reply:
Very fair point. We're going to work on a specification soon that we can use as a readme to this crate that that explicitly makes it clear what is for safety and what is for reliability / debugability.